extra/nrnivmodl-core.in

```diff
@@ -1,104 +1,121 @@
-    done < "$SAVE_FILE"
-fi
+# prefix for common options : make sure to rename these if options are changed.
+MAKE_OPTIONS="MECHLIB_SUFFIX NMODL_BINARY INCFLAGS LINKFLAGS MODS_PATH VERBOSE"
 
-# Parse args (overriding)
+# parse CLI args
 while getopts "n:m:v:d:i:l:p:hV" OPT; do
     case "$OPT" in
     n)
-        params_MECH_NAME="$OPTARG";;
+        # suffix for mechanism library
+        params_MECHLIB_SUFFIX="$OPTARG";;
     m)
+        # nmodl or mod2c binary to use
         params_NMODL_BINARY="$OPTARG";;
-    d)
-        params_DESTDIR="$OPTARG";;
     i)
+        # extra include flags
         params_INCFLAGS="$OPTARG";;
     l)
+        # extra link flags
         params_LINKFLAGS="$OPTARG";;
     V)
+        # make with verbose
         params_VERBOSE=1;;
     p)
-        PARALLELISM="$OPTARG";;
+        # option for parallel build (with -j)
+        PARALLEL_BUILDS="$OPTARG";;
     h)
         echo "$APP_NAME [options, ...] [mods_path]"
         echo "Options:"
         echo "  -n <name>       The model name, used as a suffix in the shared library"
         echo "  -m <mod2c_bin>  NMODL code generation compiler path"
         echo "  -i <incl_flags> Definitions passed to the compiler, typically '-I dir..'"
         echo "  -l <link_flags> Definitions passed to the linker, typically '-Lx -lylib..'"
-        echo "  -d <dest_dir>   Install to dest_dir. Default: off - Only build in {ARCH}"\
-             "        Consider using this option on a second call. Params are preserved."
         echo "  -V              Verbose: show commands executed by make"
-        echo "  -p <n_procs>    Make parallelism. Default: 4"
+        echo "  -p <n_procs>    Number of parallel builds (Default: $PARALLEL_BUILDS)"
         exit 0;;
     ?)
         exit 1;;
     esac
 done
 
+# consume an option
 shift $(($OPTIND - 1))
 
+# only one mod files directory is supported in neuron and coreneuron
 if [ $# -gt 1 ]; then
     echo "[ERROR] $APP_NAME expects at most one mod dir. See syntax: '$APP_NAME -h' "
     exit 1
 fi
 
-
-# If defined mods dir be in $1
+# if defined mods dir be in $1
 if [ $# -eq 1 ]; then
     # Note: due to bug #712 makefile wont handle mod dir with spaces, so we let it fail here
     params_MODS_PATH=$1
 fi
+
+# warn if no mod files provided
 if [ -d $params_MODS_PATH ]; then
-    # warn if no mod files provided
-    ls $params_MODS_PATH/*.mod || echo "WARNING: No mods found!"
+    ls $params_MODS_PATH/*.mod &>/dev/null || echo "WARNING: No mod files found, compiling default ones only!"
 else
     echo "FATAL: Invalid mods directory: '$params_MODS_PATH'"
     exit 1
 fi
 
-# Output options & populate make_params
-mkdir -p "$(dirname "$SAVE_FILE")"
-echo "# nrnivmodl-core options. Mods location: $params_MODS_PATH" > $SAVE_FILE
+# temporary directory where mod files will be copied
+temp_mod_dir=@CMAKE_HOST_SYSTEM_PROCESSOR@/core/mods
+mkdir -p $temp_mod_dir
 
+# copy mod files with include files. note that ${ROOT_DIR}/share
```

Comment:
The idea is we prepare the mod files here so that the makefile can be simpler?

Btw, for reference, maybe we can update https://github.com/BlueBrain/CoreNeuron/pull/130#issuecomment-461378302

Reply:
@ferdonline : main motivation for recompiling all mod files is to have flexibility in building special. For example, once CoreNeuron is installed, we want to compile mod files with `mod2c` or `nmodl` or different nmodl options or different compiler flags.

If inbuilt mod files are built inside `libcoreneuron`, one has to completely re-install CoreNeuron with different flags of `nmodl`. This is quite restrictive for new usage with NMODL. And hence, we no longer want to compile mod files inside `libcoreneuron`.

Reply:
> Btw, for reference, maybe we can update https://user-images.githubusercontent.com/915100/53173802-ad701780-35e8-11e9-9d28-0441aecbf517.png (in #133 (comment))

Could you provide the source of that? :)